server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -292,6 +292,46 @@ protected FieldMapper clone() {
         }
     }
 
+    @Override
+    public void validate(DocumentFieldMappers mappers) {
+        if (this.copyTo() != null && this.copyTo().copyToFields().isEmpty() == false) {
```

Comment:
We've done this sort of thing a lot in the aggs framework and it has led to some weird methods, `doValidate`, `innerValidate`, `doInnerValidate` and stuff like that. I really wish there were a clean way to force an overrides to call super. Its not something you can force with checkstyle. At least, I don't think it is.

Reply:
I've made `validate` final and added a `doValidate()` subclassable method.

Reply:
It makes me sad, but if @jtibshirani prefers it that is ok with me.

Reply:
I do prefer it as it's more robust. Unfortunately I can't think of a cleaner way to handle this. Would smoothing over the naming make you a bit less sad, perhaps using `validateOptions` instead of `doValidate`?

Reply:
> Would smoothing over the naming make you a bit less sad, perhaps using `validateOptions` instead of `doValidate`?

The name doesn't really matter. I'll know in my heart that we're doing it because we can't force subclasses to call the superclass implementation. It's cool. Compromise is a way of life.